objcxx_eh.h

```diff
@@ -10,10 +10,12 @@ extern "C" {
  * __cxa__allocate_exception(). Since the standard version is defined with
  * _GLIBCXX_NOTHROW, clang gives a type mismatch error.
  */
-#if !__cplusplus || !USE_STD_EXCEPTION_TYPES
 __attribute__((weak))
-void *__cxa_allocate_exception(size_t thrown_size);
+void *__cxa_allocate_exception(size_t thrown_size)
+#if defined __cplusplus && defined CXA_ALLOCATE_EXCEPTION_IS_NOEXCEPT
```

Comment:
I'm sorry but I'm not 100% sure what you mean here. It is a single -D flag from line 383 of `CMakeLists.txt`. The C++ check is necessary without it `eh_personality.c` fails to compile since it gets a `noexcept` in the C code. But I can make it a bit more readable by typing the function declaration twice, if you would prefer that, like so:

```
#if defined __cplusplus && defined CXA_ALLOCATE_EXCEPTION_IS_NOEXCEPT
__attribute__((weak))
void  *__cxa_allocate_exception(size_t thrown_size) noexcept;
#else
__allocate((weak))
void *__cxa_allocate_exception(size_t thrown_size);
#endif
```

Reply:
Flag should be either `-DFOO=` or `-DFOO=noexcept`, and then use `FOO` in the code (where `FOO` is a more sensible name, such as `CXA_ALLOCATE_EXCEPTION_NOEXCEPT`)